### DISCUSSION TITLE
i want auth With OAuth2 not create new record

### DISCUSSION THREAD

**Discussion by SoniSuciadi** [pocketbase/pocketbase#1979] [Q&A] [answered]
const data = await dbPocketBase
      .collection("users")
      .authWithOAuth2(
        "google",
        code,
        codeVerifier,
        "http://localhost:3000/cms/sign-in",
        {}
      );

**Comment by ganigeorgiev** [comment, marked as answer]
There is not enough context to help you debug it.

You can find OAuh2 web integration example in https://pocketbase.io/docs/authentication/#web-oauth2-integration.

If the above doesn't help, please provide a more detailed code sample (or better - a minimal reproducible repo).

**Reply by SoniSuciadi** [reply]
this created record after login with oauth,
i want after unregistered user login with oauth,
pocketbase does not create new record data

**Reply by ganigeorgiev** [reply]
I'm sorry, I don't understand what do you mean. The above seems like the success response of the OAuth2 auth call.
Once authenticated and linked to a PocketBase user we'll not create a new auth record on the next Google sign-in for the same account, if that's what you want.
You can "unlink" the already linked OAuth2 user by clicking on the user from the Admin UI and then select the tab "Authorized providers".

**Reply by ganigeorgiev** [reply]
If you want to use the OAuth2 integration without creating an auth record in PocketBase, then this is not possible at the moment via the web APIs.
